### PR TITLE
feat(table): Add table for ARC 104 onwards (#2835)

### LINKED ISSUE #2835: [Feat] テーブル「ARC104〜」を追加しましょう

### PR DESCRIPTION
**Pull request by KATO-Hiro**
close #2835

## Summary by CodeRabbit

* **New Features**
  * Added "ARC 104 Onwards" contest group for filtering and viewing ARC contests from round 104 onward
  * Expanded contest task database with many new ARC entries (rounds 104–209)

* **Tests**
  * Added comprehensive test suite covering ARC 104 onwards patterns, table generation, headers, and edge cases
  * Introduced consolidated test data for ARC 104+ scenarios

* **Documentation**
  * Added a detailed testing plan outlining test cases, mock data, and verification steps

### CODE REVIEW

File: src/lib/utils/contest_table_provider.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ import type { TaskResults, TaskResult } from '$lib/types/task';
 
 import { classifyContest, getContestNameLabel } from '$lib/utils/contest';
 import { getTaskTableHeaderName } from '$lib/utils/task';
+import { arc } from '@/test/lib/utils/test_cases/contest_type';
```
[the remark this answers is not in the document]
Comment (KATO-Hiro) [reply src/lib/utils/contest_table_provider.ts]: 該当部分は使用していないことを確認したため、削除しました
